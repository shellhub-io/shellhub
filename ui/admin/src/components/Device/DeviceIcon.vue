<template>
  <i :class="iconName" style="font-size: 20px" />
</template>

<script lang="ts">
import { computed, defineComponent } from "vue";

export default defineComponent({
  props: {
    icon: {
      type: String,
      required: true,
    },
  },
  setup(props) {
    const deviceIcon = {
      alpine: "fl-alpine",
      arch: "fl-archlinux",
      centos: "fl-centos",
      coreos: "fl-coreos",
      debian: "fl-debian",
      devuan: "fl-devuan",
      elementary: "fl-elementary",
      fedora: "fl-fedora",
      freebsd: "fl-freebsd",
      gentoo: "fl-gentoo",
      linuxmint: "fl-linuxmint",
      mageia: "fl-mageia",
      manjaro: "fl-manjaro",
      mandriva: "fl-mandriva",
      nixos: "fl-nixos",
      opensuse: "fl-opensuse",
      rhel: "fl-redhat",
      sabayon: "fl-sabayon",
      slackware: "fl-slackware",
      ubuntu: "fl-ubuntu",
      raspbian: "fl-raspberry-pi",
      "ubuntu-core": "fl-ubuntu",
      void: "fl-void",
    };

    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    const iconName = computed(() => deviceIcon[props.icon]);

    return {
      iconName,
    };
  },
});
</script>
