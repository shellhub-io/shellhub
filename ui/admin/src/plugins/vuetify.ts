// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";
import { VNumberInput } from "vuetify/labs/VNumberInput";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
  components: {
    VNumberInput,
  },
});
