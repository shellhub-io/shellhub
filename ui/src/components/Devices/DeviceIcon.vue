<template>
  <i
    data-test="type-icon"
    :class="iconName + ' mr-1'"
    style="font-size: 20px"
  />
</template>

<script setup lang="ts">
import { computed } from "vue";

const props = defineProps({
  icon: {
    type: String,
    required: true,
  },
});

const deviceIcon = {
  alpine: "fl-alpine",
  arch: "fl-archlinux",
  centos: "fl-centos",
  coreos: "fl-coreos",
  debian: "fl-debian",
  devuan: "fl-devuan",
  elementary: "fl-elementary",
  fedora: "fl-fedora",
  freebsd: "fl-freebsd",
  gentoo: "fl-gentoo",
  linuxmint: "fl-linuxmint",
  mageia: "fl-mageia",
  manjaro: "fl-manjaro",
  mandriva: "fl-mandriva",
  nixos: "fl-nixos",
  opensuse: "fl-opensuse",
  rhel: "fl-redhat",
  sabayon: "fl-sabayon",
  slackware: "fl-slackware",
  ubuntu: "fl-ubuntu",
  raspbian: "fl-raspberry-pi",
  "ubuntu-core": "fl-ubuntu",
  ubuntucore: "fl-ubuntu",
  void: "fl-void",
  docker: "fl-docker",
};

const iconName = computed(() => deviceIcon[props.icon] || "fl-tux");
</script>
