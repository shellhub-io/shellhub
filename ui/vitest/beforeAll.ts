import MockAdapter from "axios-mock-adapter";
import { beforeAll } from "vitest";
import ResizeObserver from "resize-observer-polyfill";
import { systemApi } from "../src/api/http";

const mockedSystemInfo = {
  authentication: {
    local: false,
    saml: true,
  },
  endpoints: {
    api: "localhost:80",
    ssh: "localhost:22",
  },
  setup: true,
  version: "v0.18.3",
};

const systemApiMock = new MockAdapter(systemApi.getAxios());
systemApiMock.onGet("http://localhost:3000/info").reply(200, mockedSystemInfo);

global.CSS = { supports: () => false } as never;
global.ResizeObserver = ResizeObserver;

beforeAll(() => {
  global.CSS = {
    supports: () => false,
    escape: (str: string) => str,
  } as never;
});
