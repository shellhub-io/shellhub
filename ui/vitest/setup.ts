/**
 * Vitest setup function
 */
export async function setup() {
  global.CSS = {
    supports: () => false,
    escape: (str: string) => str,
  } as never;
}

// FAIL LOUDLY on unhandled promise rejections / errors
process.on("unhandledRejection", (reason) => {
  // eslint-disable-next-line no-console
  console.log("FAILED TO HANDLE PROMISE REJECTION");
  throw reason;
});

/**
 * Vitest Teardown function
 */
export async function teardown() {
  console.log("üìù vitest globalTeardown");
}
