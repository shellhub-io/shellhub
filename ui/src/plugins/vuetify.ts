// Styles
import "@mdi/font/css/materialdesignicons.css";
import "@fortawesome/fontawesome-free/css/all.css";
import "vuetify/styles";
import { VFileUpload, VFileUploadItem } from "vuetify/labs/VFileUpload";
import "../styles/_variables.scss";
import { aliases as faAliases, fa } from "vuetify/iconsets/fa";

// Vuetify
import { createVuetify } from "vuetify";

const light = {
  dark: false,
  colors: {
    primary: "#667acc",
    secondary: "#FFFFFF",
    background: "#F5F5F5",
    tabs: "#F5F5F5",
    foreground: "#F5F5F5",
    paymentForm: "#F5F5F5",
    terminal: "#0f1526",
    "v-theme-background": "#FFFFFF",
    "v-theme-surface": "#FFFFFF",
    "v-theme-card": "#FFFFFF",
  },
};

const dark = {
  dark: true,
  colors: {
    primary: "#667acc",
    secondary: "#1E2127",
    background: "#18191B",
    tabs: "#1E1E1E",
    foreground: "#1E1E1E",
    paymentForm: "#E0E0E0",
    terminal: "#0f1526",
    "v-theme-background": "#1E2127",
    "v-theme-surface": "#1E2127",
    "v-theme-card": "#22252B",
  },
};

export default createVuetify({
  components: {
    VFileUpload,
    VFileUploadItem,
  },
  theme: {
    defaultTheme: "dark",
    themes: {
      dark,
      light,
    },
  },
  defaults: {
    global: {

    },
    VTextField: {
      variant: "outlined",
      density: "comfortable",
    },
    VSelect: {
      variant: "outlined",
      density: "comfortable",
    },
    VTextarea: {
      variant: "outlined",
    },
  },
  icons: {
    faAliases,
    sets: {
      fa,
    },
  },
});
