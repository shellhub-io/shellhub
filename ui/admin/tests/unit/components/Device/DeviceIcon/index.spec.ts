import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it } from "vitest";
import DeviceIcon from "../../../../../src/components/Device/DeviceIcon.vue";

describe("Device Icon", () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    const vuetify = createVuetify();

    wrapper = mount(DeviceIcon, {
      props: {
        icon: "ubuntu",
      },
      global: {
        plugins: [vuetify],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Renders the icon", () => {
    expect(wrapper.find("i").classes()).toContain("fl-ubuntu");
    expect(wrapper.find("i").exists()).toBeTruthy();
  });
});
