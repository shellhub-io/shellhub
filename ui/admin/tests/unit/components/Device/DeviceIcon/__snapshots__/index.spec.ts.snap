// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Device Icon > Renders the component 1`] = `"<i class="fl-ubuntu" style="font-size: 20px;"></i>"`;
